src/Akeneo/Connectivity/Connection/back/Infrastructure/Apps/Persistence/FindOneConnectedAppByConnectionCodeQuery.php

```diff
@@ -35,9 +35,14 @@ public function execute(string $connectionCode): ?ConnectedApp
             connected_app.certified,
             connected_app.connection_code,
             connected_app.user_group_name,
-            IF(test_app.client_id IS NULL, FALSE, TRUE) AS is_test_app
-        FROM akeneo_connectivity_connected_app connected_app
-        LEFT JOIN akeneo_connectivity_test_app test_app ON test_app.client_id = connected_app.id
+            IF(test_app.client_id IS NULL, FALSE, TRUE) AS is_test_app,
+            IF(access_token.token IS NULL AND auth_code.token IS NOT NULL, TRUE, FALSE) AS is_pending
+        FROM akeneo_connectivity_connected_app AS connected_app
+        LEFT JOIN akeneo_connectivity_test_app AS test_app ON test_app.client_id = connected_app.id
+        LEFT JOIN akeneo_connectivity_connection AS connection ON connection.code = connected_app.connection_code AND connected_app.connection_code = :connectionCode
```

Comment:
```suggestion
        LEFT JOIN akeneo_connectivity_connection AS connection ON connection.code = connected_app.connection_code
```
Redundant with the where clause or was it done for performance reasons ?

Reply:
You're right @RiffVentura. As SQL statements are executed in this order FROM > JOINS > WHERE > SELECT and as we need only one result, it is a little optimization to avoid linking too much data. I thought it was a good idea because there is a lot of left join in this query.